dlwls5201/MyArchitecture/app/build.gradle

```
@@ -42,6 +42,10 @@ dependencies {
     //kotlin
     implementation"org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlinVersion"
 
+    //koin
+    implementation "org.koin:koin-android:$koinVersion"
+    implementation "org.koin:koin-android-viewmodel:$koinVersion"
```

androidx 를 쓰신다면 androidx 용 koin 을 쓰시는 것이 좋을 것 같아요.
```suggestion
    implementation "org.koin:koin-androidx-viewmodel:$koinVersion"
```

---

수정했습니다.

commit : 920db97

---

질문이 있습니다.
- companion object를 사용하여 객체를 만드는 부분(SIngleTon)
을 제거하라 하신 이유가 koin에서 내부적으로 싱글톤으로 생성해주는데
한번 더 하는 불필요한 작업이기때문에 그런가요?

- 그렇다면 만약 나중에 협업을 할 때 누군가가 그 클래스를 koin을 사용하지 않고
직접 만들어 사용하게 된다면 문제가 발생하지 않나요??

---

> 질문이 있습니다.
> 
> * companion object를 사용하여 객체를 만드는 부분(SIngleTon)
>   을 제거하라 하신 이유가 koin에서 내부적으로 싱글톤으로 생성해주는데
>   한번 더 하는 불필요한 작업이기때문에 그런가요?

A : 본 프로젝트에 작성된 companion object 는 
외부에서 객체를 생성하기 위해서 만든 것으로 알고 있습니다.
하지만, koin 라이브러리를 통하여 사용하게 된다면, 
의존성 주입을 koin 이 해주기 때문에
companion object 에 정의된 함수를 통하여 객체를 생성하는 것은 의미가 없다고 봅니다.

> * 그렇다면 만약 나중에 협업을 할 때 누군가가 그 클래스를 koin을 사용하지 않고
>   직접 만들어 사용하게 된다면 문제가 발생하지 않나요??

A : 말씀하신 부분에 대해서는 개발자끼리 협의해서 
koin 을 적용한 프로젝트에서는 companion object 를 사용하여 객체를 생성하는 코드를 작성하지 않도록 해야 할 것으로 판단됩니다.
하지만, koin 을 사용하기 때문에 companion object 를 사용하면 안된다 라는 의미는 아닙니다.